scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -330,7 +330,13 @@ protected void receivedHelloVerifyRequest(HelloVerifyRequest message) throws Han
 	 */
 	protected void receivedServerHello(ServerHello message) throws HandshakeException {
 		// store the negotiated values
+
 		usedProtocol = message.getServerVersion();
+		if (usedProtocol.compareTo(ProtocolVersion.VERSION_DTLS_1_2) != 0) {
+			AlertMessage alert = new AlertMessage(AlertLevel.FATAL, AlertDescription.PROTOCOL_VERSION, session.getPeer());
+			throw new HandshakeException("The client only supports DTLS v1.2", alert);
```

Comment:
maybe rather something like : "The client does not support %usedProtocol but only DTLS v1.2" ?

Reply:
Done.